### ISSUE TITLE
Feature request - Badge Rankup timing vs shipment

### ISSUE THREAD

**Issue by Squelch20** [MasterHomebrewerProgram/mhp-issues#6]
Since we only have 4 shipments a year, we should track when a member ranksup, achieves stars, or achieves special awards compared to their subscription time periods.  as I only ship badges 4 times a year, many times when I see an expired membership with a badge rankup, I have no way of knowing if they ranked up before or after their membership expired.  I always err on the side of conservatism and send them a badge anyways.  it would be good to know with certainty if they're do a badge or not.

**Comment by apscherz** [comment]
could we do this through an alert? Brewer gets and Xbox style "achievement unlocked" kind of thing while on the backend we generate a badge serial number that correlates with a date. ie "Matt Westwick, Ace Amercian Aler 001" and that has their date they achieved it, tagged by when they upload the last qualifying scoresheet

**Comment by prphntm63** [comment]
Yeah I've set up the db to track this info

**Comment by prphntm63** [comment]
tl;dr - one will be able to query by "approved && !shipped" status, so when you're ready to process a load it'll tell you what's ready to go out
